InMemoryFileSys/Directory.cs

```diff
@@ -10,139 +12,124 @@ public class Directory : IDirectory
     public DateTime CreationDate { get; set; }
 
     // Public property to access the singleton instance
-    public static Directory Root => _root.Value;    
+    public static Directory Root => _root.Value;   
+    
     /// <inheritdoc/>
-    public int? Size
-    {
-        get
-        {
-            int totalSize = 0;
-            foreach (var entry in _entries)
-            {
-                if (entry is IFile file)
-                {
-                    totalSize += file.Size ?? 0;
-                }
-                else if (entry is IDirectory directory)
-                {
-                    totalSize += directory.Size ?? 0;
-                }
-            }
+    public int? Size { get; private set; }
+   
+    public string Path { get; }
 
-            return totalSize;
-        }
-    }
-    
-    private List<IFileSystemEntry> _entries { get; }
+    private readonly Dictionary<string, IFileSystemEntry> _entries;
     
     // Singleton instance
-    private static readonly Lazy<Directory> _root = new Lazy<Directory>(() => new Directory("/", new SystemClock()));
-    
+    // Singleton instance
+    private static readonly Lazy<Directory> _root = 
+        new(() => new Directory(rootName, new SystemClock(), null));
+
     private readonly IClock _clock;
 
-    private Directory(string? name, IClock clock)
+    private readonly Directory? _parentDirectory;
+    
+    private const string slash = "/";
+
+    private const string rootName = "/";
+    
+    private Directory(string? name, IClock clock, Directory? parentDirectory)
     {
         _clock = clock ?? throw new ArgumentNullException(nameof(clock));
 
         Name = name;
         CreationDate = _clock.UtcNow;
-        _entries = new List<IFileSystemEntry>();
+        
+        _parentDirectory = parentDirectory;
+
+        if (parentDirectory == null)
+        {
+            Path = name;
```

Comment:
the root Path should be equal directly to Slash, here currenlty you have name which is nullable.
How could we implement two cases:
1. if it is a root folder, then the name can be empty (I prefer that instead of having null)
2. if it is a non-root folder, then the name cannot be empty or null and so on.
?
Could you implement validation logic here?